app/build.gradle

```diff
@@ -168,3 +170,31 @@ dependencies {
     implementation 'androidx.test.espresso:espresso-idling-resource:3.5.1'
     implementation 'androidx.annotation:annotation:1.8.0'
 }
+
+tasks.named("clean") {
+    doLast {
+        delete(new File(projectDir, ".cxx"))
+    }
+}
+
+// This is needed because task "clean" depends on these tasks:
+//:app:externalNativeBuildCleanLiteDebug
+//:app:externalNativeBuildCleanLiteRelease
+//:app:externalNativeBuildCleanProDebug
+//:app:externalNativeBuildCleanProRelease
+// And these tasks expect to find build/conan/[ABI]/conan_toolchain.cmake files
+//
+// If build folder is malformed, we can't run `gradle clean` without this workaround.
+// This is because we supply our (conan's) CMAKE_TOOLCHAIN_FILE.
+// This workaround removes these "clean" task dependencies if conan_toolchain.cmake is not found (malformed build dir)
```

Comment:
How about creating an empty file at build/conan/[ABI]/conan_toolchain.cmake instead to make sure those clean-tasks execute successfully? That way we don't lose any additional functionality of those tasks (if there is any, ever). Seems a bit less hacky to me that way!

Reply:
Tried that, couldn't figure out why it doesn't work properly, CMake expects more files. Anyway another simpler workaround already in place